spectral_cubes/emission_search_cubes.md

```diff
@@ -0,0 +1,1101 @@
+                                                 calib_level = [0, 1, 2, 3, 4])  # Exclude planned observations
+```
+
+```{code-cell} ipython3
+print(f'We found {len(jwst_obstable)} JWST spectral cube observations.')
+```
+
+`Astroquery` has retrieved a wide variety of metadata for each of these observations. Let's look at the first couple rows:
+
+```{code-cell} ipython3
+jwst_obstable[0:2]
+```
+
+### 1.3 Crossmatching YSO coordinates to MAST footprints
+
++++
+
+Now, with that table as our starting point, let's match our YSO coordinates to the sky footprints of all these JWST observations. We already have everything from MAST that we need to do this, so we don't need to run another query. The footprint of an observation is in the `s_region` column of the observations table output by `astroquery`:
+
+```{code-cell} ipython3
+# For example...
+print(jwst_obstable[0]['s_region'])
+```
+
+This `s_region` denotes the vertices of a polygon in right ascension and declination. We can parse this string into an array of tuple coordinates with the custom `parse_polygon` function, which we imported from the `code_src` directory for this notebook.
+
+:::{warning}
+As currently written, the `s_region` passed into `parse_polygon` must be a single polygon. This is the case for all JWST and SOFIA spectral cube observations, but not all missions in the Fornax archives.
+:::
+
+:::{warning}
+If you replace `parse_polygon` with another function, be aware that the array of polygon vertices we later pass into `matplotlib.path.Path` must be *closed*, meaning that the last vertex is identical to the first vertex.
+:::
+
+```{code-cell} ipython3
+# For example...
+print(parse_polygon(jwst_obstable[0]['s_region']))
+```
+
+In a moment, we'll parse our polygons and use `Matplotlib`'s `Path` class to test whether any of our YSO coordinates are in each observation's sky footprint. `Path.contains_points` expects an array of test coordinates in an *(N, 2)* shape, where N is the number of coordinates to test. Let's make an array of YSO coordinates:
+
+```{code-cell} ipython3
+yso_coords = np.transpose(np.asarray((yso_table_clean['ra'], yso_table_clean['dec'])))
+print(yso_coords.shape)
+```
+
+Performing the test on all observation sky footprints without parallelizing, as in the following cell, would take quite a while (e.g., six or seven minutes on the Fornax 16-CPU server). On Fornax, if you uncomment this cell and open the "Kernel usage" tab to your right while it runs, you'll notice that you're using only a small percentage of the total CPU capacity that you have access to (e.g., less than 10% if you selected the 16-CPU server option when you launched Fornax). Feel free to click the `Interrupt the kernel` stop button on your notebook, because there's a better way!
+
+```{code-cell} ipython3
+# Since this cell is an example of what not to do,
+# we'll also omit the nuance of gnomonic projection: see the next section.
```

Comment:
pedagogically it's not the right thing to show what not to do, show what to do only and drop this code cell, it's perfectly adequate to narratively describe non-optimal ways, no need to show code for them.